src/util/errorHandling/apiError.ts

```diff
@@ -0,0 +1,49 @@
+import { HttpResponseCode } from "./";
+
+export class ApiError {
+  /**
+   * The HTTP status code of the error
+   */
+  public statusCode: HttpResponseCode;
+
+  /**
+   * Short description of the error
+   */
+  public error: string;
+
+  /**
+   * The message in the error
+   */
+  public message: any;
+
+  /**
+   * Creates a new ApiError
+   * @param _statusCode The HTTP status code of the error
+   * @param _message The reason of the error
+   */
+  constructor(_statusCode: HttpResponseCode, _message?: any) {
+    this.statusCode = _statusCode;
+    this.message = _message;
+
+    switch (this.statusCode) {
```

Comment:
Could replace this switch by defining a map where the key is the response code and the value is the error message and then just call the map.

Though you would need to also add an if for when the response code is not in the map but it should still be cleaner.

Reply:
Yeah, it would be cleaner. I just stole this from hs_hub for now. I will rework this when I get round to it. I've created another issue for improving the error handling (see #10)